xgbgen/main.go

```diff
@@ -49,16 +46,26 @@ func main() {
 	c := newContext()
 	c.Morph(xmlBytes)
 
+	outFile, err := os.Create(fmt.Sprintf("%s/%s.go", *packageName, *packageName))
+	if err != nil {
+		panic(err)
+	}
+	outFile.Truncate(0)
+	defer outFile.Close()
+	out := bufio.NewWriter(outFile)
+
 	if !*gofmt {
-		c.out.WriteTo(os.Stdout)
+		c.out.WriteTo(out)
 	} else {
 		cmdGofmt := exec.Command("gofmt")
 		cmdGofmt.Stdin = c.out
-		cmdGofmt.Stdout = os.Stdout
+		cmdGofmt.Stdout = out
 		cmdGofmt.Stderr = os.Stderr
 		err = cmdGofmt.Run()
 		if err != nil {
 			log.Fatal(err)
 		}
 	}
+
+	out.Flush()
```

Comment:
Prefer `defer out.Flush()` after the `bufio.NewWriter`?

Reply:
I did consider that and in this case it could be fine, however, due to the behavior of `out.Flush()` being a way to write data and is heavily dependent on data that would be defined after `out.Flush()` is stated in the code I think it makes more sense to not defer it.

Reply:
No longer an issue in 4760074eeb03154fee151b5bd99f64eae86d1426